rclpy/rclpy/node.py

```diff
@@ -453,6 +453,11 @@ def declare_parameters(
             # Get the values from the tuple, checking its types.
             # Use defaults if the tuple doesn't contain value and / or descriptor.
             name = parameter_tuple[0]
+            if namespace:
+                name = f'{namespace}.{name}'
+
+            # Note(jubeira): declare_parameters verifies the name, but set_parameters doesn't.
+            validate_parameter_name(name)
```

Comment:
I think we need to move a bit more code around.  In particular, if `parameter_tuple[0]` was not passed as a `string`, then we could potentially have an issue here.  I'll suggest that we move the `if not isinstance(name, str)` check from lines 464 below to right after we assign `parameter_tuple[0]` to `name`.